src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -397,11 +397,6 @@ impl<'a> Linker for GccLinker<'a> {
     }
 
     fn export_symbols(&mut self, tmpdir: &Path, crate_type: CrateType) {
-        // Symbol visibility in object files typically takes care of this.
-        if crate_type == CrateType::Executable {
-            return;
-        }
```

Comment:
This seems like a potentially very far-reaching change. What is it needed for?

Reply:
I  have now changed it such that the if condition will only be avoided if `override_export_symbols` is set to `Some`.
This is only set to `Some` when the target is  `x86_64-fortanix-unknown-sgx`. So that should limit the reach.